public/scripts/app.js

```diff
@@ -2,6 +2,502 @@ console.log("Sanity Check: JS is working!");
 
 $(document).ready(function(){
 
-// your code
+  // initial ajax call to bring back all profiles to create the SRP // might need to wrap this up into an on click event later down the road
+  $.ajax({
+    method: 'GET',
+    url: '/api/searchpage',
+  })
+  .then(function(allProfileData){
+    renderSearchResults(allProfileData)
+  })
+  .catch(function(err){
+    console.log('ERROR ON INITIAL LOAD', err)
+  });
 
+  // clicking on a search results card to render the profile
+  $('#search-results').on('click','.search-card-header', renderProfileFromSrp)
+
+  // click on the back to srp span to go back to the srp
+  $('.back-to-srp-span').on('click', reRenderSrp);
+
+  // edit button parent/child association - on the #user-profile, when you click on the it, the .edit-header is selected (which is the pencil) and the editProfileHeader function is called
+  $('#user-profile').on('click', '.edit-input-header', editProfileHeader);
+  // on save (when the save pencil icon is clicked) the user-profile is saved
+  $('#user-profile').on('click', '.edit-save-header', saveProfileHeader);
+
+  // this to for the modal to have a listenser to be opened
+  $('.add-new-user-button').on('click', function() {
+    console.log('modal clicked')
+     $('#add-user').modal();
+  });
+
+  // this runs the functionality for 'creating a new user' - the button is only shown on the SRP
+  $('.save-new-user').on('click', createNewUser);
+
+  $('#user-profile').on('click', '.delete-profile', removeProfile);
 });
+
+let createNewUser = function() {
+
+  // go and get each of the value from the input field. The initial value of each field was populated automatically by using the 'value=' in the input tag of the html
+  let newProfileHeaderName = $('.modal-input-for-profile-name').val();
+  let newProfileHeaderTitle = $('.modal-input-for-profile-title').val();
+  let newProfileHeaderWorkpalce = $('.modal-input-for-profile-workPlace').val();
+  let newProfileHeaderQuote = $('.modal-input-for-profile-quote').val();
+  let newProfileHeaderImg ='/images/userimages/' +  $('.modal-input-for-file').val().split('\\').pop();
```

Comment:
doesn't work for actually uploading images: you don't send the image file data to the server